user-profile/src/components/CandidatePersonalInfo/CandidatePersonalInfo.js

```diff
@@ -0,0 +1,169 @@
+import { useState, useCallback, useEffect } from "react";
+import DropdownField from "../DropdownField/DropdownField";
+import InputField from "../InputField/InputField"
+import PhoneComponent from "../PhoneComponent/PhoneComponent";
+import Button from "../Button/Button";
+import Heading from "../Heading/Heading";
+import "./CandidatePersonalInfo.css";
+import { AlertMessage } from "../AlertMessage/AlertMessage";
+
+
+function CandidatePersonalInfo() {
+
+    const [firstname, setFirstname] = useState('');
+    const [lastname, setLastname] = useState('');
+    const [dob, setDob] = useState('');
+    const [gender, setGender] = useState('');
+    const [nin, setNin] = useState('');
+    const [city, setCity] = useState('');
+    const [address, setAddress] = useState('');
+    const [linkedin, setLinkedin] = useState('');
+    const [maritalstat, setMaritalstat] = useState('');
+    const [email, setEmail] = useState('dummy@email.com');
+    const [phone, setPhone] = useState('');
+
+    const [showAlert, setShowAlert] = useState(false)
+    const [alertType, setalertType] = useState('alert');
+    const [message, setMessage] = useState('')
+    const [disableNextBtn, setDisableNextBtn] = useState(true)
+
+    const [userData, setUserData] = useState(null);
+    const signupServiceUrl = '';
+
+    useEffect(() => {
+        // fetch(signupServiceUrl)
+        //     .then((response) => response.json())
+        //     .then((data) => {
+        //         console.log(data)
+        //         setUserData(null)
+        //     });
+        //
+    }, [])
+
+    const handleFirstname = useCallback(e => {
+        let newValue = e.target.validity.valid ? e.target.value : e.target.value.substr(0, e.target.value.length - 1);
+        setFirstname(newValue);
+    }, []);
+
+    const handleLastname = useCallback(e => {
+        let newValue = e.target.validity.valid ? e.target.value : e.target.value.substr(0, e.target.value.length - 1);
+        setLastname(newValue);
+    }, []);
+
+    const handleDob = useCallback(e => {
+        setDob(e.target.value);
+    }, []);
+
+    const handleGender = useCallback(e => {
+        setGender(e.target.value);
+    }, []);
+
+    const handleNin = useCallback(e => {
+        let newValue = e.target.validity.valid ? e.target.value : e.target.value.substr(0, e.target.value.length - 1);
+        setNin(newValue);
+    }, []);
+
+    const handleCity = useCallback(e => {
+        let newValue = e.target.validity.valid ? e.target.value : e.target.value.substr(0, e.target.value.length - 1);
+        setCity(newValue);
+    }, []);
+
+    const handleAddress = useCallback(e => {
```

Comment:
Well done for using `useCallback`, one thing - it's better if the callback returns the field value rather than the full event - the full event is more information than what we need from the component

Reply:
we've refactored the code so that events are not passed to the calling components.